aten/src/ATen/native/quantized/cpu/qconv.cpp

```diff
@@ -310,6 +310,53 @@ class QConv2dInt8 final : public c10::OperatorKernel {
         output_min,
         output_max);
 
+    auto input_scale = input_contig.q_scale();
+
+    // Re-quantizing the bias based on input scale and weight scale.
+    if (input_scale != pack_ptr.input_scale) {
+      // Get the original weight and adjust it to uint8 from int8
+      auto weight_contig = pack_ptr.orig_weight;
+      auto bias_fp32 = pack_ptr.bias;
+      int8_t* w_data = (int8_t*)weight_contig.data_ptr<c10::qint8>();
+      Tensor qnnp_weight = at::_empty_affine_quantized(
+          weight_contig.sizes(),
+          at::device(kCPU).dtype(kQUInt8),
+          kernel_scale,
+          kernel_zp);
+      auto* qnnp_w_data = qnnp_weight.data_ptr<c10::quint8>();
+      for (int i = 0; i < weight_contig.numel(); ++i) {
+        qnnp_w_data[i] = static_cast<c10::quint8>(w_data[i] + 128);
```

Comment:
I am missing something here. Why do we add 128 both here and in the packing step?

Reply:
The weight tensor we store in the packedWeightStruct is int8, so we need to add 128 here. 
In pre-pack since we call the qnnpack-prepack function we need to add 128 there as well before calling it.